packages/ember-htmlbars/lib/helpers/each.js

```diff
@@ -2,6 +2,151 @@ import { get } from "ember-metal/property_get";
 import { forEach } from "ember-metal/enumerable_utils";
 import normalizeSelf from "ember-htmlbars/utils/normalize-self";
 
+/**
+  The `{{#each}}` helper loops over elements in a collection. It is an extension
+  of the base Handlebars `{{#each}}` helper.
+
+  The default behavior of `{{#each}}` is to yield its inner block once for every
+  item in an array.
+
+  ```javascript
+  var developers = [{name: 'Yehuda'},{name: 'Tom'}, {name: 'Paul'}];
+  ```
+
+  ```handlebars
+  {{#each developers as |person|}}
+    {{person.name}}
+    {{! `this` is whatever it was outside the #each }}
+  {{/each}}
+  ```
+
+  The same rules apply to arrays of primitives.
+
+  ```javascript
+  var developerNames = ['Yehuda', 'Tom', 'Paul']
+  ```
+
+  ```handlebars
+  {{#each developerNames as |name|}}
+    {{name}}
+  {{/each}}
+  ```
+
+  ### {{else}} condition
+
+  `{{#each}}` can have a matching `{{else}}`. The contents of this block will render
+  if the collection is empty.
+
+  ```handlebars
+  {{#each developers as |person|}}
+    {{person.name}}
+  {{else}}
+    <p>Sorry, nobody is available for this task.</p>
+  {{/each}}
+  ```
+
+  ### Specifying an alternative view for each item
+
+  `itemViewClass` can control which view will be used during the render of each
+  item's template. The following template:
+
+  ```handlebars
+  <ul>
+    {{#each developers itemViewClass="person" as |developer|}}
```

Comment:
We should probably not doc this, as it creates poor performance and will be removed.